evalml/demos/weather.py

```diff
@@ -15,4 +17,18 @@ def load_weather():
         + evalml.__version__
     )
     X, y = load_data(filename, index=None, target="Temp")
```

Comment:
There were two dates missing from this dataset which prevented this new implementation from working properly. Because we no longer rely on the index to determine validity of the training/test separation, the `time_index` column in X has to be as we expect it to be (ordered, interval has to be detected, datetime based) for the new tools in woodwork 0.10.0 to be usable since the `DateTimeFormatDataCheck` isn't run in most use cases.